File: app/src/androidTest/java/com/jbrunton/mymovies/TestApplication.java
package com.jbrunton.mymovies;

import com.jbrunton.entities.GenresRepository;
import com.jbrunton.entities.MoviesRepository;
import com.jbrunton.mymovies.app.ApplicationDependencies;

public class TestApplication extends MyMoviesApplication {
    @Override protected ApplicationDependencies createDependencyGraph() {
        return new ApplicationDependencies() {
            @Override public MoviesRepository moviesRepository() {
                throw new RuntimeException("Stub!");
            }

            @Override public GenresRepository genresRepository() {
                throw new RuntimeException("Stub!");
            }
        };
    }
}

File: app/src/androidTest/java/com/jbrunton/mymovies/TestApplication.kt
package com.jbrunton.mymovies

import com.jbrunton.mymovies.di.ApplicationComponent
import com.jbrunton.mymovies.di.TestApplicationComponent

class TestApplication : MyMoviesApplication() {
    override fun createApplicationComponent(): ApplicationComponent =
            TestApplicationComponent(this)
}
